docs/contributing/backward-compatibility-promise.md

```diff
@@ -0,0 +1,75 @@
+# Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+After reading this promise you'll understand backward compatibility, what changes you can expect and how we plan to make changes in the future.
+
+*Note: This BC promise becomes effective with the release first non-beta release (`7.0.0`).*
```

Comment:
Maybe `non-pre relase`, because we plan to release also `rc1` if I remember correctly.

Or ... `with the first stable release`

If none of them - there is duplicit `release` word ;-)

Reply:
`stable` it is